tracks/python/exercises/darts/mentoring.md

```diff
@@ -0,0 +1,93 @@
+# Mentoring
+
+## Problem and Challenges
+
+The problem asks you to write a function which returns earned points in a single toss of a Darts game.
+The dart's location is given as a Cartesian point (an `x` and `y` value).
+Points are based on the distance from the center.
+
+## Reasonable Solution
+
+```python
+import math
+
+
+def score(x, y):
+    distance = math.sqrt(x * x + y * y)
+    if distance <= 1:
+        return 10
+    if distance <= 5:
+        return 5
+    if distance <= 10:
+        return 1
+    return 0
+```
+
+## Talking Points
+
+### Avoiding the use of `else`
+
+It can be tempting to use `elif` for the conditionals after the first one.
+However, doing so would be an "unnecessary else after return", as flagged by Pylint ([R1705][R1705]).
+
+### Minimizing the number of comparisons
+
+The distance checks can comute the upper and lower bound for each point band.
+However, the extra comparions are not needed as they are covered by the prior conditions.
+
+```python
+if 0 <= distance <= 1:
+    return 10
+if 1 < distance <= 5:
+    return 5
+if 5 < distance <= 10:
+    return 1
+```
+
+### Reducing imports
+
+The `math` import is not strictly needed.
+A clever alternative is to compute `distance = x * x + y * y` and use squared values for the point bands: 1, 25, 100.
```

Comment:
Is it?  Clever, I mean.  I do not find it clever, really, but it surely is an alternative.  I do not even find it more complicated, just approached differently.

Reply:
Yeah. Agreed. It was something that didn't occur to me and it was surprising to me when I saw it. That triggered an emotional "that's clever" response. 

I'll go with the `git commit --amend; git push --force` route here.